File: tools/tailor_cv_py/src/tailor_cv/prompt.py
from pathlib import Path

_PROMPTS_DIR = Path(__file__).parent / "prompts"


def _load(filename: str) -> str:
    return (_PROMPTS_DIR / filename).read_text(encoding="utf-8")


SYSTEM_PROMPT = _load("cv_system.txt")
USER_TEMPLATE = _load("cv_user.txt")

COVER_LETTER_SYSTEM_PROMPT = _load("cover_letter_system.txt")
COVER_LETTER_USER_TEMPLATE = _load("cover_letter_user.txt")


DEFAULT_HEADER = _PROMPTS_DIR / "default_header.md"
DEFAULT_FOOTER = _PROMPTS_DIR / "default_footer.md"


def build_user_message(master_cv: str, job_description: str) -> str:
    return USER_TEMPLATE.format(cv=master_cv, job_description=job_description)


def build_cover_letter_message(master_cv: str, job_description: str) -> str:
    return COVER_LETTER_USER_TEMPLATE.format(cv=master_cv, job_description=job_description)

File: tools/tailor_cv_ts/src/prompt.ts
import { readFileSync } from "fs";
import { join } from "path";

export const PROMPTS_DIR = join(__dirname, "..", "prompts");

function load(filename: string): string {
  return readFileSync(join(PROMPTS_DIR, filename), "utf-8");
}

export const SYSTEM_PROMPT               = load("cv_system.txt");
export const USER_TEMPLATE               = load("cv_user.txt");
export const COVER_LETTER_SYSTEM_PROMPT  = load("cover_letter_system.txt");
export const COVER_LETTER_USER_TEMPLATE  = load("cover_letter_user.txt");
export const DEFAULT_HEADER              = load("default_header.md");
export const DEFAULT_FOOTER              = load("default_footer.md");

export function buildUserMessage(masterCv: string, jobDescription: string): string {
  return USER_TEMPLATE.replace("{cv}", masterCv).replace("{job_description}", jobDescription);
}

export function buildCoverLetterMessage(masterCv: string, jobDescription: string): string {
  return COVER_LETTER_USER_TEMPLATE.replace("{cv}", masterCv).replace("{job_description}", jobDescription);
}
